tools/ncf_doc_rudder.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Usage: ./ncf_doc_rudder.py
+#
+# This is a Python module to generate documentation from generic methods in ncf to be embedded in Rudder manual
+
+import ncf 
+import requests
+import sys
+import re
+from distutils.version import StrictVersion
+from pprint import pprint
+
+# URL of the rudder_info api
+release_info_url = "http://www.rudder-project.org/release-info/rudder/"
+
+def get_min_versions():
+  "Build the dictionnary of minimal Rudder version including compatibility with a given CFEngine version"
+  min_version = {}
+  
+  try:
+    versions = requests.get(release_info_url + "versions").content.decode('ascii').splitlines()
+  except requests.exceptions.RequestException as e:
+    print(e)
+    sys.exit(1)
+
+  for rudder_version in versions:
+    try:
+      cfengine_version = requests.get(release_info_url + "versions/" + rudder_version + "/components/cfengine").content.decode('ascii')
+    except requests.exceptions.RequestException as e:
+      print(e)
+      sys.exit(1)
+    
+    if cfengine_version in min_version:
+      if StrictVersion(rudder_version) >= StrictVersion(min_version[cfengine_version]):
+        break
```

Comment:
I don't understand why you have a break statement here. I can see you want to update min_version[cfengine_version] to always have the lowest of rudder_versions that support a given cfengine_version, but this break will *exit* the loop over rudder_versions.

For example, if versions = [ 2.11, 3.0, 3.1, 3.2 ], then you will do this:

  * min_version[3.6] = 2.11
  * break (because 3.0 >= 2.11)

So min_version will only contain 3.6, but not 3.7 (for 3.2).